tuna/main.py

```diff
@@ -83,18 +83,19 @@ def read_import_profile(filename):
     # turn is parent to `_codecs`.
     entries = []
     with open(filename, "r") as f:
-        line = f.readline()
-        assert line == "import time: self [us] | cumulative | imported package\n"
-        while True:
-            line = f.readline()
-            if not line:
-                break  # EOF
-            # remove `import time:`
-            items = line[12:].strip().split("|")
+        # extract lines prefixed with "import time: ""
+        import_lines = (
+            line[len('import time: '):].rstrip('\n')
+            for line in f
+            if line.startswith('import time: ')
+        )
+        assert next(import_lines) == "self [us] | cumulative | imported package"
+        for line in import_lines:
```

Comment:
Doesn't need to be indented.

Reply:
Which?

Reply:
The `for line in import_lines` loop. Better close the file first.

Reply:
It does, because at this point only the first line of the file has actually been read- each iteration of the for loop reads the next line(s)